src/pybind/mgr/cephadm/module.py

```diff
@@ -2574,6 +2580,41 @@ def _create_rgw(self, rgw_id, host):
     def apply_rgw(self, spec):
         return self._apply(spec)
 
+    def add_iscsi(self, spec):
+        return self._add_daemon('iscsi', spec, self._create_iscsi, self._config_iscsi)
+
+    def _config_iscsi(self, spec):
+        api_secure = 'false' if spec.api_secure is None else spec.api_secure
+        igw_conf = [
+            '# generated by cephadm',
+            '[config]',
+            'pool = {}'.format(spec.pool),
+            'trusted_ip_list = {}'.format(spec.trusted_ip_list or ''),
+            'minimum_gateways = 1',
+            'fqdn_enabled = {}'.format(spec.fqdn_enabled or ''),
+            'api_port = {}'.format(spec.api_port or ''),
+            'api_user = {}'.format(spec.api_user or ''),
+            'api_password = {}'.format(spec.api_password or ''),
+            'api_secure = {}\n'.format(api_secure)]
+
+        return {'iscsi_config': '\n'.join(igw_conf)}
+
+    @async_map_completion
+    def _create_iscsi(self, igw_id, host, extra_config):
+        ret, keyring, err = self.mon_command({
+            'prefix': 'auth get-or-create',
+            'entity': 'client.iscsi.' + igw_id,
+            'caps': ['mon', 'allow *',
+                     'mgr', 'allow r',
+                     'osd', 'allow *'],
+        })
+        extra_config['iscsi_config'] += 'cluster_client_name = client.iscsi.' + igw_id
```

Comment:
Annoyingly we don't get the client name until here, but here we also don't have the spec to build the rest of the config, so this ugly split :(

Reply:
oh like i do to pass in the config, yeah good idea!